### ISSUE TITLE
Libcec tray app forcing computer to wake up from sleep 

### ISSUE THREAD

**Issue by ravinsweden** [Pulse-Eight/libcec#546]
Hi,

I switched from version 4 to 6 and everything is working as expected, except  one thing. 

When I turn off the LG TV with the remote, the computer is suspended which is great. But after around 5-10 seconds the computer starts up and turning on the TV again. This only happens when running the Cec tray app, exiting the cec tray app and doing a manual suspend from windows start menu does not automatically wake the computer from suspend after 5-10 seconds. Is this a know bug with 6.0.2?

**Comment by nibty** [comment]
I'm having the same issue. Were you able to find a solution? Downgrading works, but I have other issues with older versions.

**Comment by ravinsweden** [comment]
Yes I solved it by disabling usb power settings in the bios. For example disabling power up by pressing any usb or pci device. It seems like the usb adapter is forcing the computer to power on like a keyboard. Please try to disable any support in the bios to power on through usb or pci.

**Comment by migloo3** [comment]
I'm still having this problem with libCEC version 7.1.1.

**Comment by opdenkamp** [comment]
Apologies this was auto-closed by the stale bot while still valid — reopened and worked through.

Root cause: the adapter has a USB remote-wakeup feature that, in autonomous mode (when the host isn't actively driving it), replies `<active source>` and wakes the PC whenever the TV probes the CEC bus for its input. A TV like a Samsung (Anynet+) probes the bus the moment the PC drops off it — so putting the PC to sleep can wake it straight back up. Because this happens at the USB/firmware level while the host is asleep, no running software can prevent it; the levers are the firmware and the adapter's autonomous-mode setting.

Two changes address it:

1. **Firmware** — a wake-guard so an autonomous power-on request within a short window after the host suspends is ignored (stops the "sleep → immediate wake" loop), while a genuine later request (turning the TV back on, or selecting this source in the TV's input list) still wakes the PC. Pending a build/flash and a version bump.

2. **libCEC + cec-tray** (#716, already merged) — autonomous mode is now a user setting. In cec-tray there's a **"Allow the TV to wake this PC"** checkbox on the Power Control tab; unchecking it (and Save) writes autonomous-mode = off to the adapter eeprom, so the TV/CEC bus no longer wakes the PC. This works on already-shipped adapters without a firmware update. Trade-off: with it off, you also can't power the PC on from the TV remote.

Until a release with these is out, the standard workaround still applies: disable device wake for the adapter (`powercfg -devicedisablewake "<adapter>"`) or turn off USB wake in the BIOS.

Leaving this open until the firmware fix is built/tested and released.

**Comment by opdenkamp** [comment]
**Likely root cause identified, and a fix is landing either way.**

Based on the logs and reports, this appears to be a **USB remote-wakeup** coming from the CEC adapter's firmware rather than from cec-tray — which would explain why it persisted with all software closed, and why disabling per-device wake in Windows didn't help. When the PC goes to sleep and drops off the CEC bus, a TV that probes the bus on standby can make the adapter reply `<active source>` and, in autonomous mode, issue a USB remote-wake to the host. I think some TVs (Samsung/Anynet+ looks like a candidate) probe aggressively the moment the PC leaves the bus. Because this happens at the USB/firmware level while the host is asleep, no running software could prevent it.

Two changes are landing together to address it:

1. **Adapter firmware v13** adds a wake-guard: for a short interval right after the host suspends, autonomous power-on requests no longer wake it. A request fired the instant the PC sleeps is ignored, but a *genuine* later action — turning the TV back on, or selecting this adapter in the TV's source list — still wakes the PC as before.

2. **libcec now exposes an autonomous-mode setting** so you can turn this behaviour off entirely without reflashing. In cec-tray it's the **"Allow the TV to wake this PC"** checkbox on the Power Control tab (also available via `cec-client -am 0` and the `bAutonomousMode` config field). Unchecking it makes the adapter stay silent when the PC isn't driving it — the trade-off is that you also lose the ability to wake the PC from the TV remote.

Both ship in the upcoming **v8.0.0 release**; the **v13 firmware is bundled in that installer**, so updating will apply the firmware fix and give you the new toggle.

Thanks for the detailed report and logs — they made this much easier to narrow down.

**Comment by migloo3** [comment]
Thanks @opdenkamp this sounds great! Any idea of when we can expect v8.0.0 to release?

**Comment by opdenkamp** [comment]
@migloo3 v8.1.6 has been released with the fix included and it's available on Github: https://github.com/Pulse-Eight/libcec/releases/tag/libcec-8.1.6
